props-core/src/main/java/sh/props/Layer.java

```diff
@@ -120,10 +161,10 @@ public Registry registry() {
   @Override
   public void accept(Map<String, String> data) {
     // disallow more than one concurrent update from taking place
-    if (!this.lock.tryLock()) {
-      log.warning(() -> "Could not update layer while another update is taking place");
-    }
-
+    // TODO(mihaibojin): this has the potential to create a lot of contention, especially if using
+    //                   `OnDemandSource`s which rely on CompletableFutures to process updates, upon
+    //                   a new key registration (LoadOnDemand#registerKey)
+    this.lock.lock();
```

Comment:
for now, ensure all updates are processed, to avoid cases where a second `update` is triggered during an ongoing update, resulting in the value being skipped.

E.g.:
- Layer accepts the new data (key1+key2)
- Layer updates key1
- key1 is updated at the source and a new `Layer::accept` is triggered
- Layer drops the update (due to the previous `tryLock` impl.)
- Layer updates key2
- Layer unlocks and completes the `accept`
- but `key1` is left un-updated / pointing to a stale value